components/DiscordBar.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react'
+import styles from '../scss/discordBar.module.scss'
+type Member = {
+  id: string
+  username: string
+  status: string
+  avatar_url: string
+}
+interface discordJSON {
+  members: Member[]
+}
+interface State extends discordJSON {
+  error?: string
+}
+const ONE_MINUTE = 60 * 1000
+const Member: React.FC<{ avatar: string }> = ({ avatar }) => {
+  return (
+    <div className="d-inline-block">
+      <img src={avatar} className={`${styles['member-avatar']} mx-1`} />
+    </div>
+  )
+}
+const DiscordBar: React.FC = () => {
+  const [data, setData] = useState<State>({
+    members: []
+  })
+  const getData = async () => {
+    try {
+      const data = await fetch(
+        'https://discord.com/api/guilds/828783458469675019/widget.json'
+      )
+      const json: discordJSON = await data.json()
+      json.members
+        ? setData(json)
+        : setData({ members: [], error: 'Something went wrong ;(' })
```

Comment:
This way it's more confusing for anyone reading the code. 

On the interface you declare that the `members` field is always defined. But then you say `json` is of the type `discordJSON` and then check if `json.members` is defined or not, which from the interface you should expect that it is indeed defined. This is a bit counter intuitive.

What I suggested before is to check if the object conforms to the interface, and then set the state with the data, using the type assertion `as discordJSON` since you now know it's valid, otherwise show an error. This makes more sense because you're validating your assumptions about the data structure before assigning it.

Also, should the error message mean something different? Why not use the same one you already have in the `catch` block by throwing an error?

Another thing, prefer using `if/else` if you're not assigning the result of the ternary operator to a variable, it helps in making the code clearer. Shorter is not always better ;)

```ts
const json = await data.json()
if ('members' in json) setData(json as discordJSON)
else setData({ members: [], error: 'Something went wrong ;(' })
// or throw an error in the else
```

Reply:
You're right, I forgot to update interface again. I implemented type assertion as you suggested. 

As for different error messages, there are two ways fetching could fail. Actual network error 'error fetching data...' and response with incorrect schema (no members field). The latter gets generic error message. It's a pretty minor detail but different messages for different scenarios look a bit better.